crates/api/src/instance.rs

```diff
@@ -19,16 +19,29 @@ impl Resolver for SimpleResolver<'_> {
 }
 
 fn instantiate(
+    config: &Config,
     compiled_module: &CompiledModule,
     imports: &[Extern],
 ) -> Result<InstanceHandle, Error> {
     let mut resolver = SimpleResolver { imports };
     unsafe {
         let instance = compiled_module
-            .instantiate(&mut resolver)
+            .instantiate(
+                config.validating_config.operator_config.enable_bulk_memory,
+                &mut resolver,
+            )
             .map_err(|e| -> Error {
                 match e {
                     InstantiationError::StartTrap(trap) => Trap::from_jit(trap).into(),
+                    e @ InstantiationError::TableOutOfBounds(_)
+                    | e @ InstantiationError::MemoryOutOfBounds(_) => {
+                        let msg = e.to_string();
+                        if config.validating_config.operator_config.enable_bulk_memory {
```

Comment:
Since we're passing this boolean into the instantiation anyway, should this sort of error handling happen in the instance? I was hoping we'd be able to avoid passing the boolean all the way down, but given the change in operational semantics I don't think we'll be able to change that.